### PR TITLE
Add accurate test cases for all helpers from ActionView::Helpers::FormBuilder documentation

### PR DESCRIPTION
**Pull request by boardfish**
#83 

All examples here are lifted from https://edgeapi.rubyonrails.org/classes/ActionView/Helpers/FormBuilder.html.

Some of the examples are skipped because they rely on other view helpers or translations.

All current failures have been rectified in other PRs.

### CODE REVIEW

**Comment by boardfish** [comment]
I wrapped up the tests for fields that received relations in the given examples. This does add Active Record to the Combustion setup, so you're welcome to revert it if you'd rather not have it here. However, I think it makes a lot of sense if we want to fully ensure the form builders will be compatible with those args.

Only thing I'd do differently there is use `rspec-rails` instead of base `rspec` so that we can use transactional fixtures and not have to clean up created records manually, but again, up to you as to whether it has a place here.

**Comment by boardfish** [comment]
Thanks for the feedback, folks! I'll find the time to act on it soon and update here when I've made progress.

**Comment by Spone** [comment]
Please let us know if we can be of some help!

**Comment by boardfish** [comment]
Really sorry, this one totally slipped my mind! 😅 I'll try to wrap this up by the end of next week, but feel free to commit to the branch if it'd be helpful to you sooner.

File: spec/view_component/form/builder_spec.rb [hunk 1]
```diff
@@ -3,83 +3,146 @@
...
+    it_behaves_like "the default form builder", :hidden_field, :pass_confirm
+    it_behaves_like "the default form builder", :hidden_field, :tag_list
+    it_behaves_like "the default form builder", :hidden_field, :token
+  end
+
+  it_behaves_like "the default form builder", :label, :title
+  it_behaves_like "the default form builder", :label, :body
+  skip "This would demonstrate translations via i18n.yml" do
+    it_behaves_like "the default form builder", :label, :cost
+  end
```
[the remark this answers is not in the document]
Comment (boardfish) [reply spec/view_component/form/builder_spec.rb:91]: Putting a locales file in `spec/internal/config/locales/en.yml` didn't change the output.

File: spec/view_component/form/builder_spec.rb [hunk 2]
```diff
@@ -3,83 +3,146 @@
...
+      content_tag(:span, builder, class: "cost_label")
+    end
+    it_behaves_like("the default form builder", :label, [:cost]) do |builder|
+      content_tag(:span, builder.translation, class: [
+                    "cost_label",
+                    ("error_label" if builder.object.errors.include?(:cost))
+                  ])
+    end
+    it_behaves_like("the default form builder", :label, [:terms]) { raw('Accept <a href="/terms">Terms</a>.') }
+  end
```
[the remark this answers is not in the document]
Comment (boardfish) [reply spec/view_component/form/builder_spec.rb:111]: Yeah, these are linked to the locales files. I can't seem to get those to load.

**Comment by boardfish** [comment]
Hey folks! I've taken this a little further today. RuboCop doesn't seem to like that we're using an OpenStruct here, but I seem to remember it being necessary here for flexibility. I also couldn't get to the bottom of the situation with the translation file.

**Comment by nicolas-brousse** [comment]
Hey @boardfish thanks for your update. We will take time in the following days to have a look.

**Comment by nicolas-brousse** [comment]
@boardfish I just merged a fix for deprecation on Rails 6.1. I resolve some issues you tell us. I guess that if you do a rebase with latest changes, it may fix some of your issues.

**Comment by nicolas-brousse** [comment]
@Spone @boardfish I suggest to merge this (after rebase) as it is. And investigate about i18n and block build that doesn't works in a separate issue. Are you agree with that? 🙂

**Comment by boardfish** [comment]
Rebased. Should be good to merge now!

**Comment by nicolas-brousse** [comment]
Thanks for your work @boardfish!

**Pull request merged**